pkg/services/sqlstore/user_auth.go

```diff
@@ -29,18 +29,33 @@ func GetUserByAuthInfo(query *models.GetUserByAuthInfoQuery) error {
 	authQuery := &models.GetAuthInfoQuery{}
 
 	// Try to find the user by auth module and id first
-	if query.AuthModule != "" && query.AuthId != "" {
-		authQuery.AuthModule = query.AuthModule
-		authQuery.AuthId = query.AuthId
+	authQuery.AuthModule = query.AuthModule
+	authQuery.AuthId = query.AuthId
 
-		err = GetAuthInfo(authQuery)
-		if !errors.Is(err, models.ErrUserNotFound) {
+	err = GetAuthInfo(authQuery)
```

Comment:
This could produce `select * from user_auth where auth_module = 'generic_oauth` - same problem as we earlier discussed. Or am I missing something that has changed? I still believe `if query.AuthModule != "" && query.AuthId != "" {` is needed for the non-generic oauth use case.

Reply:
* I think the first time `GetAuthInfo` is called, should not search for the `user_id` since it doesn't yet exist, no?
* Yeah, I guess you are right, since otherwise if they are both empty, it will go and select any row from the table.